### PR TITLE
Plans: Show error notice when we can not update files

### PR DESCRIPTION
**Pull request by ebinnion**
We've got auto-config for Jetpack plans to a point where the UI looks good. We need to start working on the error messaging and edge cases. This PR is meant to start with that.

### CODE REVIEW

**Comment by ebinnion** [comment]
I just pushed up a commit that is a rough pass at a mock by @rickybanister. Here is a screenshot of what that looks like.

screen shot 2017-04-04 at 3 24 27 pm

**Comment by rickybanister** [comment]
Let's go with something like this instead:

image

Sorry to initially lead you astray @ebinnion!

The 'manual install' button should link to https://en.support.wordpress.com/setting-up-premium-services/

And the get help link should go to the same place we're currently linking to in the old autoconfig flow. 

The whole card should go above the thank you card.

File: client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx [hunk 1]
```diff
@@ -275,18 +279,85 @@ class JetpackThankYouCard extends Component {
...
+				'We are unable to install the necessary plugins on your site because it is a multi-network sites.'
+			);
+			this.trackConfigFinished( 'calypso_plans_autoconfig_error_multinetwork' );
+		}
+
+		return (
+			<div>
+					<Card className="checkout-thank-you__jetpack-error-card">
+						<h3 className="checkout-thank-you__jetpack-error-heading">
+							{ translate( 'We had trouble setting up your plan' ) }
```
Comment (a8ci18n) [inline client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'We had trouble setting up your plan.' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (ebinnion) [inline client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx]: @rickybanister We wouldn't want a period in the heading, would we?

**Comment by ebinnion** [comment]
I've pushed another commit that gets us close to @rickybanister's mock above. We may want to tweak the font styles a bit though.

screen shot 2017-04-05 at 10 32 06 am

@rickybanister also asked me to give him a list of the error copy I'm using. Here that is:

- We are unable to install the necessary plugins on your site due to lack of file permissions.
- We are unable to install the necessary plugins on your site because it has an older version of Jetpack.
- We are unable to install the necessary plugins on your site because it is not the main network site.
- We are unable to install the necessary plugins on your site because it is a multi-network sites.

**Comment by rickybanister** [comment]
> We are unable to install the W and X plugins that power Y and Z features due to your site's current file permissions settings. You must either install them manually or change your file permissions.

> We are unable to install the W and X plugins that power Y and Z features because your site has an out of date version of Jetpack. 

In this case we need to offer them a link to update Jetpack (perhaps instead of the manual install and get help buttons), even if it takes them to wp-admin. Then we should schedule an email or something to get them back in the autoconfig flow. 

> We are unable to install the W and X plugins that power Y and Z features on your site because it is part of a multi-site network, but is not the main network site.

> We are unable to install the W and X plugins that power Y and Z features on your site because it is a multi-network sites.

I don't understand that last option—there seems to be a typo. Is the problem that this is the main site? Or that they are on a host that uses multisite? I need more info. 

This is coming along nicely!

**Comment by rickybanister** [comment]
cc @ashleighaxios for copy-double-check (I offered replacement copy suggestions in my previous comment)

**Comment by rickybanister** [comment]
And just to record it, earlier it was pointed out that if autoconfig fails, we should not display the 'view your site' link, the bottom of the thank you card can just be blank.

**Comment by ashleighaxios** [comment]
I have language thoughts, but to back up a step. Are all of these error notices saying that the process is complete, but skipped the set up of some features? Or that the process is not complete because [ reason ]? Or, some combination between the types of errors?

Right now is reads like the first, but @rickybanister's suggestions sound more like the second.

**Comment by ebinnion** [comment]
> I don't understand that last option—there seems to be a typo. Is the problem that this is the main site? Or that they are on a host that uses multisite? I need more info.

There is a typo. The last word should be singular. Also, there's a bit of a difference between multisite and multi-network. A multi-network, is like a multisite of multisites, at least that's how I understand it. So, similar to multisites, we can't install a plugin in this case.

**Comment by ebinnion** [comment]
> I have language thoughts, but to back up a step. Are all of these error notices saying that the process is complete, but skipped the set up of some features? Or that the process is not complete because [ reason ]? Or, some combination between the types of errors?

In this case, the process never started because we failed one of several checks before we tried to auto-config.

The only case that has multiple reasons is the permissions one. If necessary, we could get a bit more granular if you'd like.

**Comment by rickybanister** [comment]
@ashleighaxios I suppose it's somewhat up for debate whether it partially works or not. You do indeed get access to things like Ads, Video hosting, etc, but you do not get Antispam and backups/scans because those require Akismet and VaultPress. 

If your Jetpack version is out of date you would get nothing, so we may need to further tweak the language for that scenario (which I alluded to anyway). 

I tried to make it sound like your first suggestion actually—that those specific features were not activated because the plugins could not be installed.

**Comment by rickybanister** [comment]
this closes #11776

**Comment by rickybanister** [comment]
also closes #11132

**Comment by ebinnion** [comment]
I just pushed another change which updates the error wording based on your feedback. Here is one screenshot:

screen shot 2017-04-10 at 4 40 55 pm

I'm going to work on testing the rest of the error conditions.

**Comment by ebinnion** [comment]
I pushed a few more changes that address some more feedback from @rickybanister. Specifically:

- The thank you card description can be changed, and we conditionally set this depending on whether auto-config is successful
- We don't show the "Visit your site" button any more if auto-config fails
- If auto-config fails, we displayed the failed features in a list below the card

Updated screenshots:

screen shot 2017-04-11 at 5 45 32 pm

screen shot 2017-04-11 at 5 44 58 pm

File: client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx [hunk 2]
```diff
@@ -387,31 +507,52 @@ class JetpackThankYouCard extends Component {
...
 		);
 	}
 
+	renderDescription( progress = 0 ) {
+		const { translate } = this.props;
+		if ( 100 === progress ) {
+			return translate( "You are powered up, it's time to see your site." );
+		}
+
+		return translate( "Now that we've taken care of the plan, it's time to power up your site." );
```
Comment (a8ci18n) [inline client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Now that we've taken care of the plan, it's time to see your new site.' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by rickybanister** [comment]
Looks good @ebinnion, let's get this merged so we get some folks testing the whole thing.

File: client/blocks/plan-thank-you-card/index.jsx [hunk 3]
```diff
@@ -44,6 +45,13 @@ const PlanThankYouCard = ( {
 
 		return null;
 	};
+	const renderDescription = () => {
+		if ( description ) {
```
Comment (enejb) [inline client/blocks/plan-thank-you-card/index.jsx]: Where does `description` come from? Should we pass it into the function?

File: client/blocks/plan-thank-you-card/index.jsx [hunk 4]
```diff
@@ -19,57 +19,94 @@ import ThankYouCard from 'components/thank-you-card';
...
-	);
-};
+	renderDescription() {
+		const { description, translate } = this.props;
+		if ( description ) {
+			return description;
+		}
+
+		return translate(
+			"Now that we've taken care of the plan, it's time to see your new site."
```
Comment (lezama) [inline client/blocks/plan-thank-you-card/index.jsx]: How do we now it is a "new" site?

Comment (ebinnion) [inline client/blocks/plan-thank-you-card/index.jsx]: The plans thank you card currently is only shown to new WP.com users. We override this wording for the Jetpack use case though.

File: client/blocks/plan-thank-you-card/index.jsx (hunk 3 above)
Comment (ebinnion) [inline client/blocks/plan-thank-you-card/index.jsx]: It comes in as a prop. I've updated the PlanThankYouCard after some feedback in Slack from you.

**Comment by ebinnion** [comment]
In testing, Enej pointed out an issue where I wasn't properly handling errors after an API call. That should now be fixed in the latest commit. Thanks @enejb. 🤗

**Comment by enejb** [comment]
The code look great! Since this is only visible under the dev flag. Lets merge this and fix any other bugs we find in different PRs.

:shipit:

**Pull request merged**